test/python/synthesis/test_synthesis.py

```diff
@@ -1440,6 +1440,24 @@ def test_not_rxx_equivalent(self):
             "Equivalent gate needs to take exactly 1 angle parameter.", exc.exception.message
         )
 
+    @combine(seed=range(10), name="seed_{seed}")
+    def test_correct_unitary_custom_gate(self, seed):
+        """Test synthesis with a custom controlled u equivalent gate."""
+        unitary = random_unitary(4, seed=seed)
+
+        class CustomXXGate(RXXGate):
+            """Custom RXXGate subclass that's not a standard gate"""
```

Comment:
This gate is basically the same as `RXXGate`, right?
Perhaps we can construct an `rxx_equivalent_gate` which is not a `StandardGate`?

Reply:
Yeah, I just used an `RXXGate` for simplicity since I knew it would work. It is going through the python gate path though since this is not a standard gate, but all the math works out the same as an RXXGate. I think you're right we should have some coverage of something not equivalent to a standard gate. If you have a matrix in mind we can change this to use that instead of subclassing RXXGate.

Reply:
Perhaps we can take the same matrix as `RXXGate` (or any of the other `StandardGate`s and add some constant global phase?

Reply:
Also in line 1429 we can add to the list of gates:
`     for gate in [RXXGate, RYYGate, RZZGate, RZXGate, CPhaseGate, CRZGate]:`
the gates `CRXGate` and `CRYGate` .

Reply:
Another possibility is this parametric iSWAP gate taken from https://threeplusone.com/pubs/on_gates.pdf (Section 6.6):

![image](https://github.com/user-attachments/assets/3850f99e-9ebe-43b0-aacf-93daf213cdea)

Reply:
Also in line 1429 we can add to the list of gates:
`     for gate in [RXXGate, RYYGate, RZZGate, RZXGate, CPhaseGate, CRZGate]:`
the gates `CRXGate` and `CRYGate` .

Reply:
it seems that the XY gate suggested above is not an rxx-equivalent gate